### ISSUE TITLE
Invalid datetime conversion (timezone)

### ISSUE THREAD

**Issue by M1hacka** [Infinidat/infi.clickhouse_orm#13]
Hey.
1) I set '2017-01-01 10:00:00' to field, save it to ClickHouse
2) I receive it back, and get datetime(2017, 01, 01 05:00:00).
P. s. my timezone is UTC+05:00 as you understood.
The reason is in DateTimeField:
```
    def to_db_string(self, value, quote=True):
        return escape(int(time.mktime(value.timetuple())), quote)
```
Method [time.mktime](https://docs.python.org/2/library/time.html#time.mktime) converts local time, not UTC. 
As I read from clickhouse_orm docs:
> DateTimeField. Always in UTC

Using [calendar.timegm](https://docs.python.org/2/library/calendar.html#calendar.timegm) will fix the problem, e. g.:
`calendar.timegm(value.timetuple())`

**Comment by M1hacka** [comment]
It will be always good to set or not set tzinfo In all to_python return values. The reazon is datetime.datetime(2000, 1, 1, 0, 0, 0) == datetime.datetime(2000, 1, 1, 0, 0, 0, tzinfo=pytz.utc) returns False.

**Comment by ishirav** [comment]
That's a good point, we should probably always set the timezone to UTC. I'll fix it.

**Comment by ishirav** [comment]
@M1hacka - I'm working on this issue, but there's a problem: the database always formats datetime fields according to its own system timezone. This means that we can insert a datetime in UTC, but when we get it back it will be formatted into some other timezone (which is not specified...). I need to find a way to determine the server's timezone in order to compensate, or tell the user the do "select toString(date_time_field, 'UTC')..." instead of simply "select date_time_field..."

**Comment by M1hacka** [comment]
@ishirav May be you can use [toString or toUnixTimestamp methods](https://clickhouse.yandex/reference_en.html#toDate, toDateTime) to convert time when fetching from clickhouse? Like:
`toString(created, 'UTC')`

**Comment by M1hacka** [comment]
@ishirav Pulled your code and saw timezone_in_use parameter in Field class. Isn't it better to make it optional and validate it in DateTimeField and DateField? Or it's a strange situation when I have to pass it for IntegerField or StringField...
P. s. DateField may still have conversion problems too.

May be this will help in getting timezone?
http://stackoverflow.com/questions/35085289/getting-timezone-name-from-utc-offset

**Comment by ishirav** [comment]
I might make timezone_in_use optional, as you say it's strange passing it to irrelevant field types.

But now I believe DateTimeField is always correct and always in UTC, even when the server or the client aren't. This requires a relatively new version of ClickHouse, that supports the timezone() function.

As for DateField, the only potential problem I see is if you assign a datetime value in a non-UTC timezone. It might need to be converted to UTC first, and then to a date value. Can you think of another scenario?

**Comment by M1hacka** [comment]
May be you are right about DateTime, I saw toDate(date, tz) converter in ClickHouse docs and though it may be a bug.

**Comment by ishirav** [comment]
Yeah I'll take care of it

**Comment by ishirav** [comment]
Fixed in 0.8.0
